app/src/main/java/me/martinez/sergio/daggertwobasearchitecture/di/ComponentProxyProd.java

```diff
@@ -0,0 +1,84 @@
+package me.martinez.sergio.daggertwobasearchitecture.di;
+
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivity;
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivityComponent;
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivityComponentProd;
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivityModule;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.AppComponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.AppComponentProd;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.DaggerAppComponentProd;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.SectionComponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.modules.ActivityModule;
+import me.martinez.sergio.daggertwobasearchitecture.di.modules.FragmentModule;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragmentComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragmentModule;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.TestSectionModule;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.firststep.FirstFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.firststep.FirstFragmentComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.firststep.FirstFragmentModule;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.secondstep.SecondFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.secondstep.SecondFragmentComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.secondstep.SecondFragmentModule;
+
+public class ComponentProxyProd implements ComponentProxy {
```

Comment:
I completely agree. This is the part I like the least too. Hopefully, I can use the fork of the project as a playground to find a work around.
That idea of yours sounds worth a try, by all means.